File: app/src/main/java/com/kyleriedemann/giantbombvideoplayer/Models/Image.java
package com.kyleriedemann.giantbombvideoplayer.Models;

import com.google.gson.annotations.SerializedName;

import org.parceler.Parcel;

/**
 * Created by kyle on 3/31/15.
 */
@Parcel
public class Image {

    @SerializedName("icon_url")
    String iconUrl;

    @SerializedName("medium_url")
    String mediumImageUrl;

    @SerializedName("screen_url")
    String screenUrl;

    @SerializedName("small_url")
    String smallImageUrl;

    @SerializedName("super_url")
    String superImageUrl;

    @SerializedName("thumb_url")
    String thumbImageUrl;

    @SerializedName("tiny_url")
    String tinyImageUrl;

    public Image(String iconUrl) {
    }

    public String getIconUrl() {
        return iconUrl;
    }

    public void setIconUrl(String iconUrl) {
        this.iconUrl = iconUrl;
    }

    public String getMediumImageUrl() {
        return mediumImageUrl;
    }

    public void setMediumImageUrl(String mediumImageUrl) {
        this.mediumImageUrl = mediumImageUrl;
    }

    public String getScreenUrl() {
        return screenUrl;
    }

    public void setScreenUrl(String screenUrl) {
        this.screenUrl = screenUrl;
    }

    public String getSmallImageUrl() {
        return smallImageUrl;
    }

    public void setSmallImageUrl(String smallImageUrl) {
        this.smallImageUrl = smallImageUrl;
    }

    public String getSuperImageUrl() {
        return superImageUrl;
    }

    public void setSuperImageUrl(String superImageUrl) {
        this.superImageUrl = superImageUrl;
    }

    public String getThumbImageUrl() {
        return thumbImageUrl;
    }

    public void setThumbImageUrl(String thumbImageUrl) {
        this.thumbImageUrl = thumbImageUrl;
    }

    public String getTinyImageUrl() {
        return tinyImageUrl;
    }

    public void setTinyImageUrl(String tinyImageUrl) {
        this.tinyImageUrl = tinyImageUrl;
    }

    @Override
    public String toString() {
        return "Image{" +
                "iconUrl='" + iconUrl + '\'' +
                ", mediumImageUrl='" + mediumImageUrl + '\'' +
                ", screenUrl='" + screenUrl + '\'' +
                ", smallImageUrl='" + smallImageUrl + '\'' +
                ", superImageUrl='" + superImageUrl + '\'' +
                ", thumbImageUrl='" + thumbImageUrl + '\'' +
                ", tinyImageUrl='" + tinyImageUrl + '\'' +
                '}';
    }
}

File: app/src/main/java/com/kyleriedemann/giantbombvideoplayer/Video/Models/Image.kt
package com.kyleriedemann.giantbombvideoplayer.video.models

import android.os.Parcelable
import androidx.room.ColumnInfo
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Image(
        @ColumnInfo(name = "icon_url") @SerializedName("icon_url") var iconUrl: String,
        @ColumnInfo(name = "medium_url") @SerializedName("medium_url") var mediumImageUrl: String,
        @ColumnInfo(name = "screen_url") @SerializedName("screen_url") var screenUrl: String,
        @ColumnInfo(name = "small_url") @SerializedName("small_url") var smallImageUrl: String,
        @ColumnInfo(name = "super_url") @SerializedName("super_url") var superImageUrl: String,
        @ColumnInfo(name = "thumb_url") @SerializedName("thumb_url") var thumbImageUrl: String,
        @ColumnInfo(name = "tiny_url") @SerializedName("tiny_url") var tinyImageUrl: String
): Parcelable
